sw/host/spiflash/verilator_spi_interface.cc

```diff
@@ -91,21 +92,34 @@ bool VerilatorSpiInterface::Init() {
   return true;
 }
 
-bool VerilatorSpiInterface::TransmitFrame(const uint8_t *tx, uint8_t *rx,
-                                          size_t size) {
+bool VerilatorSpiInterface::TransmitFrame(const uint8_t *tx, size_t size) {
   size_t bytes_written = write(fd_, tx, size);
   if (bytes_written != size) {
     std::cerr << "Failed to write bytes to spi interface. Bytes written: "
               << bytes_written << " expected: " << size << std::endl;
     return false;
   }
   usleep(kWriteReadDelay);
-  size_t bytes_read = ReadBytes(fd_, rx, size);
+  return true;
+}
+
+bool VerilatorSpiInterface::CheckHash(const uint8_t *tx, size_t size) {
+  uint8_t hash[SHA256_DIGEST_LENGTH];
+  SHA256_CTX sha256;
+  SHA256_Init(&sha256);
+  SHA256_Update(&sha256, tx, size);
+  SHA256_Final(hash, &sha256);
+
+  uint8_t *rx = (uint8_t *)malloc(size);
+  ssize_t bytes_read = ReadBytes(fd_, rx, size);
   if (bytes_read < size) {
     std::cerr << "Failed to read bytes from spi interface. Bytes read: "
               << bytes_read << " expected: " << size << std::endl;
   }
-  return true;
+  bool hash_matches = !std::memcmp(rx, hash, SHA256_DIGEST_LENGTH);
```

Comment:
if using a container, you can return the result of this function directly,.

Reply:
Done.